### PR TITLE
(BIDS-2248) update epochs with missing slot status

### PR DESCRIPTION
**Pull request by MauserBitfly**
### <samp>🤖 Generated by Copilot at 6941bef</samp>

Added a new command `expot-epoch-missed-slots` to `cmd/misc/main.go` to export rewards data for epochs with missing slot blocks. This helps to monitor and evaluate validator behavior and incentives.

### CODE REVIEW

**Review by D13ce — CHANGES_REQUESTED** [review]
Please see my comment, functionality you use is deprecated and has been replaced. I have linked an example that should show you how to upgrade to the new functionality.

Please adapt this PR before I continue my review.

File: cmd/misc/main.go [hunk 1]
```diff
@@ -155,6 +155,46 @@ func main() {
...
 			err = exporter.ExportEpoch(epoch, rpcClient)
 
+			if err != nil {
+				logrus.Errorf("error exporting epoch: %v", err)
+			}
+			logrus.Printf("finished export for epoch %v", epoch)
+		}
+	case "export-epoch-missed-slots":
+		logrus.Infof("exporting epochs with missed slots")
+		err = services.InitLastAttestationCache(utils.Config.LastAttestationCachePath)
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply cmd/misc/main.go]: InitLastAttestationCache removed

File: cmd/misc/main.go [hunk 2]
```diff
@@ -149,6 +149,42 @@ func main() {
...
+			)
+			SELECT e.epoch FROM (
+				SELECT (slot / $1) AS epoch
+				FROM blocks
+				WHERE status = '0' 
+				AND slot < (SELECT slot FROM last_exported_epoch)
+				ORDER BY slot desc
+			) AS e 
+			GROUP BY e.epoch 
+			ORDER BY epoch;
```
Comment (LuccaBitfly) [inline cmd/misc/main.go:175]:
```suggestion
			WITH last_exported_epoch AS (
			    SELECT (MAX(epoch) * $1) AS slot
			    FROM epochs
			    WHERE finalized AND rewards_exported
			)
			SELECT epoch
			FROM blocks
			WHERE status = '0' AND slot < (SELECT slot FROM last_exported_epoch)
			GROUP BY epoch
			ORDER BY epoch;
```

**Review by LuccaBitfly — CHANGES_REQUESTED** [review]

File: cmd/misc/main.go (hunk 2 above)
Comment (MauserBitfly) [reply cmd/misc/main.go]: you are right the query was more complex then needed - simplified now.

**Review by LuccaBitfly — APPROVED** [review]
lgtm, pls don't forget to get a review from @peterbitfly

File: cmd/misc/main.go [hunk 3]
```diff
@@ -151,6 +151,39 @@ func main() {
...
+		}
+	case "export-epoch-missed-slots":
+		logrus.Infof("exporting epochs with missed slots")
+		epochs := []uint64{}
+		err = db.ReaderDb.Select(&epochs, `
+			WITH last_exported_epoch AS (
+				SELECT (MAX(epoch)*$1) AS slot 
+				FROM epochs 
+				WHERE finalized 
+				AND rewards_exported
```
Comment (peterbitfly) [inline cmd/misc/main.go:167]: what is the reason to filter on `rewards_exported` ?

Comment (MauserBitfly) [reply cmd/misc/main.go:167]: To get the last epoch that was exported. If the epoch has not yet been exported we don't need to re-trigger an export.

**Review by peterbitfly — APPROVED** [review]

**Review by D13ce — APPROVED** [review]

**Pull request merged**